### PR TITLE
Remove old task runners and futures modules

### PR DESCRIPTION
**Pull request by desertaxle**
Moves `new_task_runners` and `new_futures` to `task_runners` and `futures`, respectively. Deletes old `task_runners` and `future` modules.

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] If this pull request adds new functionality, it includes unit tests that cover the changes
- [ ] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed.

For new functions or classes in the Python SDK:

- [ ] This pull request includes helpful docstrings.
- [ ] If a new Python file was added, this pull request contains a stub page in the Python SDK docs and an entry in `mkdocs.yml` navigation.

### CODE REVIEW

File: scripts/wait-for-server.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
 
 async def main(timeout):
-    async with anyio.move_on_after(timeout):
+    with anyio.move_on_after(timeout):
```
Comment (desertaxle) [inline scripts/wait-for-server.py:27]: This fixes the failures we've been seeing with the benchmarks job

**Review by abrookins — APPROVED** [review]
Looks good to me. 👍

**Comment by abrookins** [comment]
One question I have that isn't directly about this PR but that it reminded me of is why did we remove the SequentialTaskRunner? I get that it doesn't serve a purpose in production code, but I thought it was there to help people write tests for their flows easier. 🤔

**Comment by desertaxle** [comment]
> One question I have that isn't directly about this PR but that it reminded me of is why did we remove the SequentialTaskRunner? I get that it doesn't serve a purpose in production code, but I thought it was there to help people write tests for their flows easier. 🤔

I removed it because the `SequentialTaskRunner` breaks the model of `.submit` triggering a non-blocking execution of a task. I would suggest the `ThreadPoolTaskRunner` to users if they want to test their flows since it maintains non-blocking execution, but you can run it locally without any dependencies.

**Pull request merged**